osquery/tables/ipmi/ipmi_info.cpp

```diff
@@ -0,0 +1,1297 @@
+
+  getPARM(lp, IPMI_LANPARM_SET_IN_PROGRESS, key, "set_in_progress", client);
+}
+
+void IPMIFullyUpCB(ipmi_domain_t* domain, void* data) {
+  TLOG << "OpenIPMI is now fully up";
+  IPMIClient* c = reinterpret_cast<IPMIClient*>(data);
+  c->setDomain(domain);
+
+  return;
+}
+
+/// Gets the the value suffix for a sensor reading.
+std::string getSensorThresholdSuffix(ipmi_sensor_t* sensor) {
+  std::string percent, base, mod_use, modifier, rate;
+
+  base = ipmi_sensor_get_base_unit_string(sensor);
+
+  if (ipmi_sensor_get_percentage(sensor)) {
+    percent = "%";
+  }
+
+  switch (ipmi_sensor_get_modifier_unit_use(sensor)) {
+  case IPMI_MODIFIER_UNIT_NONE:
+    break;
+
+  case IPMI_MODIFIER_UNIT_BASE_DIV_MOD:
+    mod_use = "/";
+    modifier = ipmi_sensor_get_modifier_unit_string(sensor);
+    break;
+
+  case IPMI_MODIFIER_UNIT_BASE_MULT_MOD:
+    mod_use = "*";
+    modifier = ipmi_sensor_get_modifier_unit_string(sensor);
+    break;
+  }
+
+  rate = ipmi_sensor_get_rate_unit_string(sensor);
+
+  return percent + " " + base + mod_use + modifier + rate;
+}
+
+void readThresholdSensorCB(ipmi_sensor_t* sensor,
+                           int err,
+                           enum ipmi_value_present_e value_present,
+                           unsigned int raw_value,
+                           double val,
+                           ipmi_states_t* states,
+                           void* data) {
+  if (err != 0) {
+    LOG(ERROR) << "Could not read sensor: " << strerror(err);
+    return;
+  }
+
+  Row r;
+
+  ipmi_mc_t* mc = ipmi_sensor_get_mc(sensor);
+  r["mc_name"] = getMCName(mc, "unknown");
+  r["mc_id"] = INTEGER(ipmi_mc_device_id(mc));
+
+  const int maxChar = 256;
+  char name[maxChar];
+  ipmi_sensor_get_name(sensor, name, maxChar) < 1 ? r["name"] = "missing"
+                                                  : r["name"] = name;
+
+  r["sensor_type"] = ipmi_sensor_get_sensor_type_string(sensor);
+
+  switch (value_present) {
+  case IPMI_NO_VALUES_PRESENT:
+    r["value"] = "no reading available";
+    break;
+
+  case IPMI_RAW_VALUE_PRESENT:
+    r["value"] = std::to_string(raw_value);
+    break;
+
+  default:
+    r["value"] = std::to_string(val) + getSensorThresholdSuffix(sensor);
+  }
+
+  r["threshold_out_of_range"] =
+      (ipmi_is_threshold_out_of_range(states, IPMI_LOWER_NON_CRITICAL) ||
+       ipmi_is_threshold_out_of_range(states, IPMI_UPPER_NON_CRITICAL))
+          ? "1"
+          : "0";
+
+  IPMIClient* c = reinterpret_cast<IPMIClient*>(data);
+  c->insertRowsQueue(r["name"] + "-" + r["mc_name"] + "-" + r["mc_id"], r);
+}
+
+void getThresholdSensorCB(ipmi_entity_t* entity, void* data) {
+  ipmi_entity_iterate_sensors(
+      entity,
+      [](ipmi_entity_t* ent, ipmi_sensor_t* sensor, void* data) {
+        if (ipmi_sensor_get_event_reading_type(sensor) ==
+            IPMI_EVENT_READING_TYPE_THRESHOLD) {
+          int rv = ipmi_sensor_get_reading(sensor, readThresholdSensorCB, data);
```

Comment:
Sure.  But `int` is short to type haha.  

As a general rule for the code base, should we juts be using auto whenever possible then?  There are quite a few more places where this would be relevant then if that were the case..